[Validator] Add "Translatable Objects" example (#19054)
symfony/symfony-docs: validation/translations.rst

You may need to clear your cache (even in the dev environment) after creating
this file for the first time.

> You can also use :class:`Ns\\TranslatableMessage` to build your violation message::

Reviewer: ```suggestion
You can also use :class:`Symfony\\Component\\Translation\\TranslatableMessage` to build your violation message::
```

`Ns` in my example was just a placeholder for the real namespace :slightly_smiling_face:

Author: Ok
I though you gave a shortcut or something

    use Symfony\Component\Translation\TranslatableMessage;
    use Symfony\Component\Validator\Constraints as Assert;
    use Symfony\Component\Validator\Context\ExecutionContextInterface;
    
    #[Assert\Callback]
    public function validate(ExecutionContextInterface $context, mixed $payload): void
    {
        // somehow you have an array of "fake names"
        $fakeNames = [/* ... */];
    
        // check if the name is actually a fake name

>         if (in_array($this->getFirstName(), $fakeNames)) {

Reviewer: ```suggestion
        if (in_array($this->getFirstName(), $fakeNames, true)) {
```

Let's be strict where we can :smile:

this file for the first time.

Translatable Objects
--------------------

In some cases, you may want to translate content from a simple function with multiple checks or contexts.
For example, when using the ``Callback`` constraints.

You can use the :class:`Symfony\\Component\\Translation\\TranslatableMessage` class like this:

.. configuration-block::

    .. code-block:: php-attributes

        use Symfony\Component\Translation\TranslatableMessage;
        use Symfony\Component\Validator\Constraints as Assert;
        use Symfony\Component\Validator\Context\ExecutionContextInterface;
        
        #[Assert\Callback]
        public function validate(ExecutionContextInterface $context, mixed $payload): void
        {
            // somehow you have an array of "fake names"
            $fakeNames = [/* ... */];
        
            // check if the name is actually a fake name
            if (in_array($this->getFirstName(), $fakeNames)) {
                $context->buildViolation(new TranslatableMessage('author.name.fake', [], 'validators'))
                    ->atPath('firstName')
                    ->addViolation()
                ;
            }
        }

> Learn more about 

Author: ```suggestion
Learn more about
```
